crates/bevy_reflect/src/type_name.rs

```diff
@@ -0,0 +1,77 @@
+use std::borrow::Cow;
+
+/// Provide the name of the type as string.
+pub trait TypeName {
+    fn name() -> Cow<'static, str>;
+}
+
+/// A object-safe version of [`TypeName`].
+/// Automatically implemented via blanket implementation.
```

Comment:
Nit: This should probably have a newline before to separate the paragraphs.
```suggestion
/// 
/// Automatically implemented via blanket implementation.
```